### PR TITLE
Port common_planning_interface_objects

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

This's essential for `moveit_ros_visualization` port, but it's blocked by planning_scene_monitor

TODO:
- [x] Port package.xml

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the MIGRATION.md notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp [hunk 1]
```diff
@@ -52,14 +54,16 @@ struct SharedStorage
...
   boost::mutex lock_;
+  rclcpp::Clock::SharedPtr clock_;
   std::weak_ptr<tf2_ros::Buffer> tf_buffer_;
   std::map<std::string, robot_model::RobotModelWeakPtr> models_;
-  std::map<std::string, CurrentStateMonitorWeakPtr> state_monitors_;
+  // TODO(JafarAbdi): uncomment once PSM is merged
+  // std::map<std::string, CurrentStateMonitorWeakPtr> state_monitors_;
 };
 
 SharedStorage& getSharedStorage()
```
Comment (JafarAbdi) [inline moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: I think this needs a port to modern c++

File: moveit_ros/planning_interface/package.xml [hunk 2]
```diff
@@ -15,28 +15,32 @@
   <url type="bugtracker">https://github.com/ros-planning/moveit/issues</url>
   <url type="repository">https://github.com/ros-planning/moveit</url>
 
-  <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend>ament_cmake</buildtool_depend>
   <buildtool_depend>python-catkin-pkg</buildtool_depend>
```
Comment (henningkayser) [inline moveit_ros/planning_interface/package.xml]: Remove

File: moveit_ros/planning_interface/common_planning_interface_objects/include/moveit/common_planning_interface_objects/common_objects.h [hunk 3]
```diff
@@ -44,19 +46,8 @@ namespace planning_interface
...
-  &robot_model, const std::shared_ptr<tf2_ros::Buffer>& tf_buffer,
-    ros::NodeHandle nh = ros::NodeHandle() ). It calls this function using the default constructed ros::NodeHandle
-
-  @param robot_model
-  @param tf_buffer
-  @return
- */
-planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot_model::RobotModelConstPtr& robot_model,
-                                                                     const std::shared_ptr<tf2_ros::Buffer>& tf_buffer);
+robot_model::RobotModelConstPtr getSharedRobotModel(const rclcpp::Node::SharedPtr node,
```
Comment (henningkayser) [inline moveit_ros/planning_interface/common_planning_interface_objects/include/moveit/common_planning_interface_objects/common_objects.h]:
```suggestion
robot_model::RobotModelConstPtr getSharedRobotModel(const rclcpp::Node::SharedPtr& node,
```
Same for others

File: moveit_ros/planning_interface/common_planning_interface_objects/include/moveit/common_planning_interface_objects/common_objects.h [hunk 4]
```diff
@@ -66,9 +57,8 @@ planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot
   @param nh A ros::NodeHandle to pass node specific configurations, such as callbacks queues.
```
Comment (henningkayser) [inline moveit_ros/planning_interface/common_planning_interface_objects/include/moveit/common_planning_interface_objects/common_objects.h]: Update documentation

File: moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp [hunk 5]
```diff
@@ -37,6 +37,7 @@
 #include <moveit/common_planning_interface_objects/common_objects.h>
 #include <moveit/robot_model_loader/robot_model_loader.h>
 #include <tf2_ros/transform_listener.h>
+#include <boost/thread/mutex.hpp>
```
Comment (henningkayser) [inline moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: Where is the mutex used?

File: moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp [hunk 6]
```diff
@@ -105,15 +106,17 @@ std::shared_ptr<tf2_ros::Buffer> getSharedTF()
   std::shared_ptr<tf2_ros::Buffer> buffer = s.tf_buffer_.lock();
   if (!buffer)
   {
-    tf2_ros::Buffer* raw = new tf2_ros::Buffer();
+    s.clock_ = std::make_shared<rclcpp::Clock>();
```
Comment (mlautman) [inline moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: You may want to explicitly define the clock type with RCL_ROS_TIME

Comment (henningkayser) [reply moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: Done

**Pull request merged**